test/lolex-test.js

```diff
@@ -2084,6 +2084,24 @@ describe("lolex", function () {
                 delete Performance.prototype.someFunc2;
                 delete Performance.prototype.someFunc3;
             });
+
+            it("should replace the getEntries, getEntriesByX methods with noops that return []", function () {
+                Performance.prototype.getEntries =
+                Performance.prototype.getEntriesByName =
+                Performance.prototype.getEntriesByType = function () { return ["foo"]; };
+
+                this.clock = lolex.install();
+
+                assert.equals(performance.getEntries(), []);
+                assert.equals(performance.getEntriesByName(), []);
+                assert.equals(performance.getEntriesByType(), []);
+
+                this.clock.uninstall();
+
+                delete Performance.prototype.getEntries;
```

Comment:
I am just wondering about the cleanup bit. Will the original functions now be "restored"?

Reply:
Just added a test that the original functions are restored, it does appear to work.

I also have a test for Cypress that runs in actual Chrome that shows the restoration working IRL: https://github.com/cypress-io/cypress/pull/4108/files#diff-c0a131a0b58412db68e166e9d7a16383R75

Reply:
I _think_ he asked about the restoration within tests since you override and `delete` from `prototype`

Reply:
Hmm, I don't see what you mean. Is there something else that needs to be tested?

Reply:
dunno, maybe something like

```js
const origGetEntries = Performance.prototype.getEntries;
const origGetEntriesByName = Performance.prototype.getEntriesByName;
const origGetEntriesByType = Performance.prototype.getEntriesByType; =
```

then restore them instead of the `delete`? not sure. If the `Performance` object is already a fake, then it doesn't matter 🙂

Reply:
I could be wrong, but it looks like `lolex` just uses the prototype to know what methods to override on the global `performance`, so I don't think anything additional needs to be tested here.

Reply:
@fatso83 is this ready?

Reply:
Sure. If you and Ben thinks this is fine, go ahead

Reply:
I think this is fine :]